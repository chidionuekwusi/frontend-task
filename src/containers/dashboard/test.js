import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Dashboard from "./index";
import NotesContext from "../../contexts/notes";

describe("<Dashboard />", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <NotesContext.Provider
        value={{
          get: () => {
            return [{ title: "Fake", content: "Content", id: "1" }];
          }
        }}
      >
        <Dashboard />
      </NotesContext.Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders all notes", () => {
    const component = mount(
      <NotesContext.Provider
        value={{
          get: () => {
            return [
              { title: "Fake", content: "Content", id: "1" },
              { title: "Fake 2", content: "Content 2", id: "2" },
              { title: "Fake 3", content: "Content 2", id: "3" }
            ];
          }
        }}
      >
        <Dashboard />
      </NotesContext.Provider>
    );
    expect(component.find(".fas.fa-trash").length).toBe(3);
  });

  it("can search for note", () => {
    const component = mount(
      <NotesContext.Provider
        value={{
          get: () => {
            return [
              { title: "Fake", content: "Content", id: "1" },
              { title: "supreme", content: "Content 2", id: "2" },
              { title: "Fake 3", content: "Content 2", id: "3" }
            ];
          }
        }}
      >
        <Dashboard />
      </NotesContext.Provider>
    );
    component
      .find("input")
      .simulate("change", { target: { value: "supreme" } });
    expect(component.find(".fas.fa-trash").length).toBe(1);
  });
});
