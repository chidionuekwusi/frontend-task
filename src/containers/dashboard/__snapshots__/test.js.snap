// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div>
  <div>
    <i
      className="fas fa-search"
    />
    <input
      id="query"
      onChange={[Function]}
      placeholder="Search for a note"
      value=""
    />
  </div>
  <div>
    <div
      onClick={[Function]}
    >
      <p>
        Fake
      </p>
      <span>
        Content
      </span>
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    <i
      className="fas fa-plus"
    />
  </button>
</div>
`;
